test/common/http/async_client_impl_test.cc

```diff
@@ -40,7 +41,7 @@ class AsyncClientImplTest : public testing::Test {
     message_->headers().insertMethod().value(std::string("GET"));
     message_->headers().insertHost().value(std::string("host"));
     message_->headers().insertPath().value(std::string("/"));
-    ON_CALL(*cm_.conn_pool_.host_, zone()).WillByDefault(ReturnRef(local_info_.zoneName()));
+    ON_CALL(*cm_.conn_pool_.host_, zone()).WillByDefault(ReturnRefOfCopy(local_info_.zoneName()));
```

Comment:
This overrides [`HostDescription::zone()`](
https://github.com/lyft/envoy/blob/2cf029ca44a2238a90460cf5269dbc342e87d24f/include/envoy/upstream/host_description.h#L76), i.e.:

    virtual const std::string& zone() const PURE;

not [`LocalInfo::zoneName()`](https://github.com/lyft/envoy/blob/2cf029ca44a2238a90460cf5269dbc342e87d24f/include/envoy/local_info/local_info.h#L28).